src/core/unit.c

```diff
@@ -3917,7 +3924,23 @@ int unit_deserialize(Unit *u, FILE *f, FDSet *fds) {
                                 log_unit_debug(u, "Failed to parse cgroup-invalidated-mask %s, ignoring.", v);
                         continue;
 
-                } else if (streq(l, "ref-uid")) {
+                } else if (streq(l, "allow-bind-bpf-link-fd")) {
+                        int fd;
+
+                        if (safe_atoi(v, &fd) < 0 || fd < 0 || !fdset_contains(fds, fd))
+                                log_unit_debug(u, "Failed to parse allow-bind-bpf-link-fd value: %s", v);
+                        else {
+                                if (fdset_remove(fds, fd) < 0) {
+                                         log_unit_debug(u, "Failed to remove allow-bind-bpf-link-fd %d from fdset", fd);
+
+                                         continue;
+                                }
+
+                                (void) allow_bind_restore(u, fd);
```

Comment:
why the indirection? why not fill the fd directly into `u->ipv4_allow_bind_bpf_link` again?

Reply:
@poettering 

libbpf needs an additional helper to construct a link directly from fd: https://github.com/systemd/systemd/pull/17655#discussion_r594588152